src/code/z_actor.c

```diff
@@ -457,439 +1245,3495 @@ f32 Actor_XZDistanceToPoint(Actor* actor, Vec3f* point) {
+                                                  transitionActorList->pos.z, 0, rotY, 0,
+                                                  (i << 0xA) + (transitionActorList->params & 0x3FF),
+                                                  transitionActorList->rotY & 0x7F, 0x3FF, 0) != NULL) {
+                    transitionActorList->id = -transitionActorList->id;
+                }
+                numTransitionActors = globalCtx->doorCtx.numTransitionActors;
+            }
+        }
+    }
+}
+
+Actor* Actor_SpawnEntry(ActorContext* actorCtx, ActorEntry* actorEntry, GlobalContext* globalCtx) {
+    s16 rotX = (actorEntry->rot.x >> 7) & 0x1FF;
+    s16 rotY = (actorEntry->rot.y >> 7) & 0x1FF;
+    s16 rotZ = (actorEntry->rot.z >> 7) & 0x1FF;
+
+    if (!(actorEntry->id & 0x8000)) {
+        rotY *= 0x10000 / 360.0f;
+    } else if (rotY > 180) {
+        rotY -= 360;
+    }
+
+    if (!(actorEntry->id & 0x4000)) {
+        rotX *= 0x10000 / 360.0f;
+    } else if (rotX > 180) {
+        rotX -= 360;
+    }
+
+    if (!(actorEntry->id & 0x2000)) {
+        rotZ *= 0x10000 / 360.0f;
+    } else if (rotZ > 180) {
+        rotZ -= 360;
+    }
+
+    return Actor_SpawnAsChildAndCutscene(actorCtx, globalCtx, actorEntry->id & 0x1FFF, actorEntry->pos.x,
+                                         actorEntry->pos.y, actorEntry->pos.z, rotX, rotY, rotZ,
+                                         actorEntry->params & 0xFFFF, actorEntry->rot.y & 0x7F,
+                                         ((actorEntry->rot.x & 7) << 7) | (actorEntry->rot.z & 0x7F), NULL);
+}
+
+Actor* Actor_Delete(ActorContext* actorCtx, Actor* actor, GlobalContext* globalCtx) {
+    s32 pad;
+    Player* player = GET_PLAYER(globalCtx);
+    Actor* newHead;
+    ActorOverlay* overlayEntry = actor->overlayEntry;
+
+    if ((player != NULL) && (actor == player->unk_730)) {
+        func_80123DA4(player);
+        Camera_ChangeMode(Play_GetCamera(globalCtx, Play_GetActiveCameraIndex(globalCtx)), 0);
+    }
+
+    if (actor == actorCtx->targetContext.arrowPointedActor) {
+        actorCtx->targetContext.arrowPointedActor = NULL;
+    }
+
+    if (actor == actorCtx->targetContext.unk8C) {
+        actorCtx->targetContext.unk8C = NULL;
+    }
+
+    if (actor == actorCtx->targetContext.bgmEnemy) {
+        actorCtx->targetContext.bgmEnemy = NULL;
+    }
+
+    func_801A72CC(&actor->projectedPos);
+    Actor_Destroy(actor, globalCtx);
+
+    newHead = Actor_RemoveFromCategory(globalCtx, actorCtx, actor);
+    ZeldaArena_Free(actor);
+
+    if (overlayEntry->vramStart != NULL) {
+        overlayEntry->numLoaded--;
+        Actor_FreeOverlay(overlayEntry);
+    }
+
+    return newHead;
+}
+
+s32 func_800BB59C(GlobalContext* globalCtx, Actor* actor) {
+    s16 x;
+    s16 y;
+
+    Actor_GetScreenPos(globalCtx, actor, &x, &y);
+
+    return (x > -20) && (x < gScreenWidth + 20) && (y > -160) && (y < gScreenHeight + 160);
```

Comment:
This makes sense, but I haven't named it like this because of the extra 20 pixels(?) horizontally and 160 pixels vertically this function is considering. So, this is checking on the screen and around the screen maybe?

Reply:
I think this is just margin for 3D actors because coords is a bad way to detect if an actor is in view on the camera.

Hypothetically, if you had a humanoid actor like Anju, where is her exact position? Like, she has a large 3D shape body and a collider, but her position is still a 3D coordinate dot in space. Is it her feet, for floor bg and gravity or is it her head where you z target her?

Wherever you put her single position dot, there will be times where her model is on screen and that dot is not. If the dot is in her feet like I suspect, then just standing behind her, looking away from her, will cut off her feet only showing her head because of camera angle. Even if you put the dot in the dead center, C-UP viewing the face will cut off the dot.

I figured it gives more vertical room and some horizontal room for these reasons, that most actors are taller than wide, and to help catch if an actor by the position only (not some camera geometry bg check thing which would be more expensive and is probably what they used for pictobox)

Reply:
Could change the numbers and see if it does anything, in the asm if necessary.